client/signup/steps/p2-complete-profile/index.jsx

```diff
@@ -0,0 +1,260 @@
+					},
+					true
+				);
+			}
+		} else {
+			initialState = {
+				fullName: {
+					value: '',
+				},
+			};
+		}
+
+		this.formStateController = new formState.Controller( {
+			fieldNames: [ 'fullName' ],
+			sanitizerFunction: this.sanitize,
+			validatorFunction: this.validate,
+			onNewState: ( state ) => {
+				this.setState( { form: state } );
+			},
+			onError: this.handleFormControllerError,
+			debounceWait: VALIDATION_DELAY_AFTER_FIELD_CHANGES,
+			hideFieldErrorsOnChange: true,
+			initialState: initialState,
+			skipSanitizeAndValidateOnFieldChange: true,
+		} );
+
+		this.state = {
+			form: this.formStateController.getInitialState(),
+			isSubmitting: false,
+			suggestedSubdomains: [],
+			lastInvalidSite: '',
+		};
+	}
+
+	componentWillUnmount() {
+		this.save();
+	}
+
+	validate = ( fields, onComplete ) => {
+		const messages = {};
+
+		if ( isEmpty( fields.fullName ) || fields.fullName.length < 3 ) {
+			messages.fullName = {
+				[ ERROR_CODE_MISSING_FULL_NAME ]: this.props.translate(
+					'Please enter your full name (3 characters or more).'
+				),
+			};
+		}
+
+		onComplete( null, messages );
+	};
+
+	handleSubmit = ( event ) => {
+		event.preventDefault();
+
+		this.setState( { isSubmitting: true } );
+
+		this.formStateController.handleSubmit( ( hasErrors ) => {
+			const fullName = formState.getFieldValue( this.state.form, 'fullName' );
+
+			if ( hasErrors ) {
+				this.setState( { isSubmitting: false } );
+
+				return;
+			}
+
+			recordTracksEvent( 'calypso_signup_p2_complete_profile_step_submit' );
+
+			// API calls
+
+			this.props.saveUserSettings( { display_name: fullName } );
+
+			const stepData = {
+				stepName: this.props.stepName,
+				form: this.state.form,
+				fullName,
+			};
+
+			this.props.submitSignupStep( stepData );
+
+			this.props.goToNextStep();
+		} );
+	};
+
+	handleFormControllerError = ( error ) => {
+		if ( error ) {
+			throw error;
+		}
+	};
+
+	save = () => {
+		this.props.saveSignupStep( {
+			stepName: 'p2-complete-profile',
+			form: this.state.form,
+		} );
+	};
+
+	handleBlur = () => {
+		this.formStateController.sanitize();
+		this.save();
+	};
+
+	handleChangeEvent = ( event ) => {
+		this.formStateController.handleFieldChange( {
+			name: event.target.name,
+			value: event.target.value,
+		} );
+	};
+
+	getErrorMessages = ( fieldName ) => {
+		const messages = formState.getFieldErrorMessages( this.state.form, fieldName );
+
+		if ( ! messages ) {
+			return;
+		}
+
+		return messages;
+	};
+
+	formFields = () => {
+		const fieldDisabled = this.state.isSubmitting;
+
+		return (
+			<>
+				<ValidationFieldset
+					errorMessages={ this.getErrorMessages( 'fullName' ) }
+					className="p2-complete-profile__validation-full-name"
+				>
+					<FormLabel htmlFor="full-name-input">
+						{ this.props.translate( 'Your Full Name' ) }
+					</FormLabel>
+					<FormTextInput
+						id="full-name-input"
+						autoFocus={ true } // eslint-disable-line jsx-a11y/no-autofocus
```

Comment:
Auto-focusing the fields is not recommended for a good reason. Are we certain we really need it here?

In any case, `={ true }` is redundant:

```suggestion
						autoFocus // eslint-disable-line jsx-a11y/no-autofocus
```

Although I'd vote for removing this prop completely.

Reply:
Will do.